neuromation/api/storage.py

```diff
@@ -713,3 +743,22 @@ def leave(self, data: StorageProgressLeaveDir) -> None:
 
     def fail(self, data: StorageProgressFail) -> None:
         pass
+
+
+def retries(msg: str) -> Iterator[ContextManager[None]]:
+    sleeptime = 0.0
+    for r in range(ATTEMPTS)[::-1]:
+        if r:
+
+            @contextlib.contextmanager
+            def retry() -> Iterator[None]:
```

Comment:
Isn't the function should be *async*?
IIUC an exception raised during the work (not the first coroutine call but in the middle of the coroutine unwinding) is not handled.

It would be nice to have a test for `retries()` to check this.

Reply:
You are right. There is `asyncio.sleep()` inside, so the context manager should be asynchronous.